pkg/drainer/drainer.go

```diff
@@ -28,50 +28,59 @@ import (
 
 var drainHelper *drain.Helper
 var node *corev1.Node
+var nthConfig config.Config
 
-//InitDrainer will ensure the drainer has all the resources necessary to complete a drain
-func InitDrainer(nthConfig config.Config) {
+// InitDrainer will ensure the drainer has all the resources necessary to complete a drain.
+// This function must be called before any function in the drainer package.
+func InitDrainer(inputNthConfig config.Config) {
+	nthConfig = inputNthConfig
 	if drainHelper == nil {
-		drainHelper = getDrainHelper(nthConfig)
+		drainHelper = getDrainHelper()
 	}
-	node = &corev1.Node{}
 	var err error
-	if !nthConfig.DryRun {
-		node, err = drainHelper.Client.CoreV1().Nodes().Get(nthConfig.NodeName, metav1.GetOptions{})
-		if err != nil {
-			log.Fatalf("Couldn't get node %q: %s\n", nthConfig.NodeName, err.Error())
-		}
-		log.Printf("Successufully retrieved node: %s", node.Name)
+	node, err = FetchNode()
+	if err != nil {
+		log.Fatalf("Couldn't get node %q: %s\n", nthConfig.NodeName, err.Error())
```

Comment:
FYI, you don't need to use `Error()` :) You can just pass err to the formatting function and the `%s` marker will be interpolated (the error's `String()` method is called. Alternately, you can use the `%w` interpolation marker for Go1.13+ to bring in any wrapped error details.

Reply:
ah good to know, I removed the Error() in drainer. There's probably more places that this should be simplified, but I'll get to that in a cleanup PR.